File: widgetssdk/src/main/java/com/glia/widgets/core/engagement/domain/model/ChatMessageInternal.java
package com.glia.widgets.core.engagement.domain.model;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.glia.androidsdk.Operator;
import com.glia.androidsdk.chat.Chat;
import com.glia.androidsdk.chat.ChatMessage;

import java.util.Optional;

public class ChatMessageInternal {
    @NonNull
    private final ChatMessage chatMessage;
    @Nullable
    private final Operator operator;

    public ChatMessageInternal(@NonNull ChatMessage chatMessage, @Nullable Operator operator) {
        this.chatMessage = chatMessage;
        this.operator = operator;
    }

    public ChatMessageInternal(ChatMessage chatMessage) {
        this(chatMessage, null);
    }

    @NonNull
    public ChatMessage getChatMessage() {
        return chatMessage;
    }

    public Optional<Operator> getOperator() {
        return Optional.ofNullable(operator);
    }

    public Optional<String> getOperatorId() {
        return getOperator().map(Operator::getId);
    }

    public Optional<String> getOperatorName() {
        return getOperator().map(Operator::getName);
    }

    public Optional<String> getOperatorImageUrl() {
        return getOperator().map(Operator::getPicture).flatMap(Operator.Picture::getURL);
    }

    public boolean isNotVisitor() {
        return getChatMessage().getSenderType() != Chat.Participant.VISITOR;
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/engagement/domain/model/ChatMessageInternal.kt
package com.glia.widgets.core.engagement.domain.model

import com.glia.androidsdk.chat.Chat
import com.glia.androidsdk.chat.ChatMessage
import com.glia.widgets.core.engagement.data.LocalOperator

internal data class ChatMessageInternal(val chatMessage: ChatMessage, val operator: LocalOperator? = null) {
    val operatorId: String? get() = operator?.id
    val operatorName: String? get() = operator?.name
    val operatorImageUrl: String? get() = operator?.imageUrl
    val isNotVisitor: Boolean get() = chatMessage.senderType != Chat.Participant.VISITOR
}
